src/formatOnSave.test.js

```diff
@@ -49,6 +51,21 @@ test('it executes prettier on buffer range if file is whitelisted regardless of
   expect(executePrettierOnBufferRange).toHaveBeenCalledWith(editor, rangeFixture);
 });
 
+test('it does not execute prettier if whitelist is provided and file is not whitelisted', () => {
+  // $FlowFixMe
+  helpers.isWhitelistProvided.mockImplementation(() => true);
+  // $FlowFixMe
+  helpers.isInScope.mockImplementation(() => true);
+  // $FlowFixMe
+  helpers.isFilePathExcluded.mockImplementation(() => false);
+  // $FlowFixMe
+  helpers.isFilePathWhitelisted.mockImplementation(() => false);
+
+  formatOnSave(editor, filePathFixture);
+
+  expect(executePrettierOnBufferRange).toHaveBeenCalledTimes(0);
```

Comment:
Great question. I do prefer it because I actually like to have that assertion for every one of my mock functions I care about. Rather than just assuming that a function was called once, I like to be explicit about how many times it's called. So I prefer `toHaveBeenCalledTimes` over `toHaveBeenCalled` in general, and I extend that to when it's not called as well :)